extensions/openshift-client/runtime/src/main/java/io/quarkus/it/openshift/client/runtime/OpenShiftClientProducer.java

```diff
@@ -2,47 +2,37 @@
 
 import jakarta.annotation.PreDestroy;
 import jakarta.enterprise.inject.Produces;
+import jakarta.enterprise.inject.Specializes;
 import jakarta.inject.Singleton;
 
-import org.jboss.logging.Logger;
-
 import io.fabric8.kubernetes.client.Config;
-import io.fabric8.kubernetes.client.KubernetesClientBuilder;
 import io.fabric8.kubernetes.client.utils.KubernetesSerialization;
 import io.fabric8.openshift.client.OpenShiftClient;
-import io.fabric8.openshift.client.OpenShiftConfig;
 import io.quarkus.arc.DefaultBean;
+import io.quarkus.kubernetes.client.runtime.KubernetesClientProducer;
 
 @Singleton
-public class OpenShiftClientProducer {
-
-    private static final Logger LOGGER = Logger.getLogger(OpenShiftClientProducer.class);
+public class OpenShiftClientProducer /* extends KubernetesClientProducer */ { // If we keep the extends, then the parent's class method is registered as an extension
 
-    private OpenShiftClient client;
+    // Temporary delegation workaround to illustrate how the behavior should be
+    private KubernetesClientProducer delegate;
 
+    @Specializes // Annotation is ignored (at class level and method level). Both, the parent's class method producer and
     @DefaultBean
     @Singleton
     @Produces
-    public OpenShiftClient openShiftClient(KubernetesSerialization kubernetesSerialization,
-            Config config) {
-        // TODO - Temporary fix for https://github.com/fabric8io/kubernetes-client/pull/3347 + WithOpenShiftTestServer
-        final OpenShiftConfig openShiftConfig = new OpenShiftConfig(config);
-        openShiftConfig.setHttp2Disable(config.isHttp2Disable());
-
-        client = new KubernetesClientBuilder()
-                .withConfig(openShiftConfig)
-                .withKubernetesSerialization(kubernetesSerialization)
-                .build()
-                .adapt(OpenShiftClient.class);
-
-        return client;
+    //    @Override
+    public OpenShiftClient kubernetesClient(KubernetesSerialization kubernetesSerialization, Config config) {
+        delegate = new KubernetesClientProducer();
```

Comment:
Can't this be injected by CDI?

Reply:
Yes, that's the previous commit with the named bean approach -[full previous discussion](https://github.com/quarkusio/quarkus/pull/35525#discussion_r1304156833)-.

However, then we get the bean ambiguity when injecting a `KubernetesClient` (not OpenShift) that doesn't contain a `@Named` annotation, thus breaking the `BeanDefaultsTest`.
This can then be fixed by removing the `@DefaultBean` annotation in the `OpenShiftClientProducer`. However, then we will break the `BeanOverridesTest` (unless we further annotate the bean overrides with priority or something else).

The main problem is that if we want to preserve the frictionless approach and not introduce breaking changes to close #35383, then we need to follow the specialization approach or something along those lines.